core/src/calleeFunctions/index.ts

```diff
@@ -64,63 +62,58 @@ export const getPools = async (
         return await routeToPool(network, calleeConfig.route, collateral.symbol);
     }
     if ('automaticRouter' in calleeConfig) {
-        const { route, fees } = await fetchAutoRouteInformation(network, collateral.symbol, amount.toFixed());
-        if (!route) {
-            throw new Error('No automatic route can be found');
+        const { pools } = await fetchAutoRouteInformation(network, collateral.symbol, amount.toFixed());
```

Comment:
- Where do we still use `getPools` if we the proposal is to never use them separately from the price?
- Why do we ever send amount as a string?

Reply:
> Where do we still use getPools if we the proposal is to never use them separately from the price?

`getCalleeData` for UniswapV3 has 2 use cases:

  - autorouter
    - the prices and pools are extracted at the same time. No need to perform other actions (`getPools` is not called)
  - hardcoded
    - the route is provided, prices are not extracted with pools (because alpharouter is not called) and we have to do the old fashioned extraction from the hardcoded route. But we also have to unify the format to use pools primarily. So the `getPools` is called for the sake of converting route to pool. 

I will refactor the code a bit so that `getPools` cannot be called for callee configs where autorouter is enabled.

Reply:
removed unused arg (amount)